pkg/command/apitoken.go

```diff
@@ -0,0 +1,105 @@
+)
+
+func newAPITokenCmd() *cobra.Command {
+	apiTokenCmd := &cobra.Command{
+		Use:     "api-token",
+		Short:   "Manage API Tokens for Tanzu Platform",
+		Aliases: []string{"apitoken"},
+		Annotations: map[string]string{
+			"group": string(plugin.SystemCmdGroup),
+		},
+	}
+
+	apiTokenCmd.SetUsageFunc(cli.SubCmdUsageFunc)
+	apiTokenCmd.AddCommand(
+		newAPITokenCreateCmd(),
+	)
+
+	return apiTokenCmd
+}
+
+func newAPITokenCreateCmd() *cobra.Command {
+	createCmd := &cobra.Command{
+		Use:     "create",
+		Short:   "Create a new API Token for Tanzu Platform",
+		Aliases: []string{},
+		Example: `
+		# Create an API Token for the Tanzu Platform
+		tanzu api-token create
+	
+		# Note: The retrieved token can be used as the value of TANZU_API_TOKEN
+		# when running 'tanzu login'. For example:
+		TANZU_API_TOKEN=<token> tanzu login`,
+		RunE:              createAPIToken,
+		ValidArgsFunction: noMoreCompletions,
+	}
+
+	return createCmd
+}
+
+func createAPIToken(cmd *cobra.Command, _ []string) (err error) {
+	c, err := config.GetActiveContext(types.ContextTypeTanzu)
+	if err != nil {
+		return errors.New("no active context of type `tanzu`. Please login to Tanzu Platform first to generate an API token")
+	}
+	if c == nil || c.GlobalOpts == nil || c.GlobalOpts.Auth.Issuer == "" {
+		return errors.New("invalid active context of type `tanzu`. Please login to Tanzu Platform first to generate an API token")
+	}
+	// Make sure it is of type tanzu with tanzuIdpType as `uaa` else return error
+	if idpType, exist := c.AdditionalMetadata[config.TanzuIdpTypeKey]; !exist || idpType != string(config.UAAIdpType) {
+		return errors.New("only UAA IDP type is supported for generating an API token. The generation of the API token for public endpoint should be done via https://console.tanzu.broadcom.com")
+	}
+
+	var token *commonauth.Token
+	// If user chooses to use a specific local listener port, use it
+	// Also specify the client ID to use for token generation
+	loginOptions := []commonauth.LoginOption{
+		commonauth.WithListenerPortFromEnv(constants.TanzuCLIOAuthLocalListenerPort),
+		commonauth.WithClientID(uaa.GetAlternateClientID()),
+	}
+
+	token, err = uaa.TanzuLogin(c.GlobalOpts.Auth.Issuer, loginOptions...)
+	if err != nil {
+		return errors.Wrap(err, "unable to login")
+	}
+
+	// Get tanzu platform endpoint as best effort from the existing context
+	tpEndpoint := "<tanzu-platform-endpoint>"
+	if hubEndpoint, exist := c.AdditionalMetadata[config.TanzuHubEndpointKey]; exist && hubEndpoint != nil {
+		u, err := url.Parse(hubEndpoint.(string))
+		if err == nil {
+			tpEndpoint = fmt.Sprintf("%s://%s", u.Scheme, u.Host)
+		}
+	}
+
+	cyanBold := color.New(color.FgCyan).Add(color.Bold)
+	bold := color.New(color.Bold)
+
+	fmt.Fprint(cmd.OutOrStdout(), bold.Sprint("==\n\n"))
+	fmt.Fprintf(cmd.OutOrStdout(), "%s Your generated API token is: %s\n\n", bold.Sprint("API Token Generation Successful!"), cyanBold.Sprint(token.RefreshToken))
```

Comment:
We may have to workshop the err/output messages in a followup. Some thoughts


- Is it better to output the token in stdout, and show the rest in stderr?
- do we want to advocate TANZU_API_TOKEN=%s tanzu login, or mention the config file way to persist the env var?

The messages in these function

- may confuse users who are interacting with the public endpoint, who ends up creating a valid `tanzu` endpoint and still get a surprise when this fails for him
- the instructions for regenerate and re-login might be confusing to the user of the current CLI instance (who has obviously logged in)

Reply:
> Is it better to output the token in stdout, and show the rest in stderr?

I would prefer using `stdout` for both the token and the rest of the output. Since this is an output message intended for the user, it's more conventional to keep all output in `stdout`. 

> do we want to advocate TANZU_API_TOKEN=%s tanzu login, or mention the config file way to persist the env var?

I feel persisting the env var in the config file is a generic way to configure any env variable. So I don't feel to strongly that we also mention the persisting the TANZU_API_TOKEN in the configuration file. Also if user is not aware of what it means to persist a setting in the config file as env.TANZU_API_TOKEN it might create even more confusion because all the follow up `tanzu login` attempt will try to use the token automatically.